### DISCUSSION TITLE
Adding MSC Apex api auto completion support

### DISCUSSION THREAD

**Discussion by edik485** [bthornton191/adams_vscode#7] [Ideas]
Is there a way to take this extension, and add support for auto completion in MSC apex api?

**Comment by bthornton191** [comment]
I would need stub files for the Apex API. I don't have time at the moment to make those manually, but let me check with the Hexagon development team. I talked to them a few years back about providing these, but I don't know the status on that request. 

Assuming we can acquire or create the stub files, yes, I could easily include them in this extension.

**Reply by edik485** [reply]
Maybe an existing Al language model can map all the functions,classes, etc into a __init__.py file from the written documentation? or atleast create the stubs?

**Reply by bthornton191** [reply]
I like that idea.  I might be able to just open up the doc folder in vscode and use copilot. I'll give that a try when I get some time. 

By the way, I'd welcome other collaborators if you wanted to take a crack at it yourself!🙂

**Reply by bthornton191** [reply]
I just gave it a shot with copilot.  I can get it to generate a stub file for a single html file, but it's missing a lot of stuff. And I can't yet get it to do the whole documentation site in bulk. 

Here's what it gave me from **_discrete_tie_8h_source.html**

```python
# DiscreteTie.pyi
from typing import Any, Protocol

class IDisplayable(Protocol):
    pass

class IUserHighlightable(Protocol):
    pass

class IUserAttributes(Protocol):
    pass

class IName(Protocol):
    def getName(self) -> str: ...
    def getPath(self) -> str: ...

class DiscreteTie(IDisplayable, IUserHighlightable, IUserAttributes, IName):
    def getType(self) -> str: ...
```

It's probably going to be easier for me to get my hands on the actual source code and generate it from that. But I'll have to track that down. 

In the mean time, let me know if you have any thoughts on how to do the LLM approach. All the info is in the documenation, so I'm sure it could be done.
